ls1intum/Artemis
# [Bugfix] Grading Configuration sum of test case weights 0

### Checklist
- [x] Client: I added client tests

### Motivation and Context


Fixes Sentry issues: 
- https://sentry.io/organizations/ls1intum/issues/2062065732/?environment=prod&project=1440029&query=is%3Aunresolved+assigned%3Ame&statsPeriod=14d
- https://sentry.io/organizations/ls1intum/issues/2062065695/?environment=prod&project=1440029&query=is%3Aunresolved+assigned%3Ame&statsPeriod=14d

### Description

- Added check on server to make sure that sum of weights is not <=0
- Added check on client to make sure that sum of weights is not <=0

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Go to Programming Exercise section and click on `Grading` button of a exercise
4. Set all weights to 0 and klick on save button
5. Check that a error alert pops up.
6. Play around with the weights, everything should work as expected 
6.1 Try also different scenarios for setting all weights to 0 e.g. First halft set to 0 then save -> should work, second half set to 0 -> should not work, or set all weights to 0 except of 1 weight and then set the last weight to 0

### Test Coverage





### Screenshots


<img width="1035" alt="image" src="https://user-images.githubusercontent.com/41108487/101561407-94a34280-39c5-11eb-8937-a04e3419a906.png">
